modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/new-content/NewContentDialogUpload.ts

```diff
@@ -0,0 +1,109 @@
+import {ContentSummaryAndCompareStatus} from 'src/main/resources/assets/js/app/content/ContentSummaryAndCompareStatus';
+import {
+    uploadDataUrlImage,
+    UploadMediaError,
+    uploadMediaFile,
+    UploadMediaSuccess,
+    uploadRemoteImage,
+} from '../../../api/uploadMedia';
+import {
+    addContentTreeUploadItem,
+    removeContentTreeUploadItem,
+    updateContentTreeUploadItemProgress,
+} from '../../../store/contentTreeUpload.store';
+import {generateUniqueName} from '../../../utils/image/generateUniqueName';
+import {reload} from '../../../store/contentTreeLoadingStore';
+import {NotifyManager} from '@enonic/lib-admin-ui/notify/NotifyManager';
+
+type UploadOptions = {
+    dataTransfer: DataTransfer;
+    parentContent?: ContentSummaryAndCompareStatus;
+};
+
+export function uploadMediaFiles({dataTransfer, parentContent}: UploadOptions) {
+    if (dataTransfer.files.length === 0) return;
+
+    const files = Array.from(dataTransfer.files);
+
+    const tasks = files.map((file) => {
+        const id = file.name;
+        const name = file.name;
+        const parentId = parentContent?.getContentId()?.toString();
+
+        addContentTreeUploadItem(id, name, parentId);
+
+        return uploadMediaFile({
+            id,
+            file,
+            parentContent,
+            onProgress: (id, progress) => updateContentTreeUploadItemProgress(id, progress),
+        });
+    });
+
+    tasks.forEach((task) => {
+        task.match(handlers.onEachSuccess, handlers.onEachError);
+    });
+}
+
+export function uploadDragImages({dataTransfer, parentContent}: UploadOptions) {
+    const htmlData = dataTransfer.getData('text/html');
+    const imgSources = extractImageSources(htmlData);
+
+    if (imgSources.length === 0) return;
+
+    const tasks = imgSources.map((src) => {
+        if (!src.startsWith('data:')) return uploadRemoteImage({imageSource: src, parentContent});
+
+        const id = src;
+        const name = generateUniqueName(src);
+        const parentId = parentContent?.getContentId()?.toString();
+
+        addContentTreeUploadItem(id, name, parentId);
+
+        return uploadDataUrlImage({
+            id,
+            name,
+            imageSource: src,
+            parentContent,
+            onProgress: (id, progress) => updateContentTreeUploadItemProgress(id, progress),
+        });
+    });
+
+    tasks.forEach((task) => {
+        task.match(handlers.onEachSuccess, handlers.onEachError);
+    });
+}
+
+//
+// * Internal
+//
+const handlers = {
+    onEachSuccess: (success: UploadMediaSuccess) => {
+        removeContentTreeUploadItem(success.mediaIdentifier);
+
+        reload();
+
+        NotifyManager.get().showSuccess(`${success.mediaIdentifier} uploaded successfully!`);
```

Comment:
Use `showSuccess` and other notifiers directly, it looks cleaner. We have it exported from `'@enonic/lib-admin-ui/notify/MessageBus'`